### PR TITLE
Sustainer: Product Alert

### PR DESCRIPTION
**Pull request by jleandroperez**
### Details
In this PR we're:

1. Implementing a Product Tier Alert, presented whenever the Sustainer Header is pressed
2. Hiding the Sidebar Sustainer Header, whenever the user is an active subscriber
3. Fixing a glitch processing refunds
4. Wiring **Tracks Events** for all the things Sustainer!

Ref. #1491

### Test: Sidebar
1. Edit the scheme and open Options
2. Make sure the **TestConfiguration.storekit** StoreKit configuration is selected
3. Launch Simplenote and log into your account
5. Open the Sidebar
6. Tap on the **Sustainer** banner

 - [ ] Verify an Alert shows up, offering the different price tiers
 - [ ] Verify that if you go ahead and purchase something, the Sidebar's Header gets hidden
 - [ ] Verify that the "Sustainer" Header does get presented in the Settings UI
 - [ ] Verify that pressing over the **Sustainer** Header (Settings UI) doesn't do anything (no alerts should show up)

### Test: Settings
1. Repeat steps 1-4 from the previous scenario
2. Press over **Settings**
3. Press over the **Sustainer** banner

- [ ] Verify the IAP flow works as expected

### Test: Refunds
1. Edit the scheme and open Options
5. Make sure the TestConfiguration.storekit StoreKit configuration is selected
7. Launch Simplenote and log into your account
8. Open the Sidebar
9. Tap on the **Sustainer** banner
10. Proceed subscribing
11. Open the **StoreKit Manager** and refund your subscription

- [ ] Verify that the **Become a Subscriber** Headers show up again
- [ ] Verify that you can become a subscriber again

### Release
These changes do not require release notes.

### CODE REVIEW

File: Simplenote/StoreManager.swift [hunk 1]
```diff
@@ -91,11 +95,26 @@ class StoreManager {
...
                 NSLog("[StoreManager] Purchase Failed \(error)")
             }
         }
     }
+
+    /// Returns the Display Price for a given Product
+    ///
+    func displayPrice(for storeProduct: StoreProduct) -> String? {
+        guard let product = storeProductMap[storeProduct] else {
+            return nil
```
Comment (jleandroperez) [inline Simplenote/StoreManager.swift:109]: This should **never** happen (AFAIK)

File: Simplenote/TagListViewController.swift [hunk 2]
```diff
@@ -576,15 +596,19 @@ private extension TagListViewController {
     }
 
     func refreshTableHeaderView() {
-        guard let headerView = tableView.tableHeaderView as? SustainerView else {
+        guard let sustainerHeaderView else {
+            return
+        }
+
+        if isActiveSustainer {
+            tableView.tableHeaderView = nil
```
Comment (jleandroperez) [inline Simplenote/TagListViewController.swift:604]: We're, now, hiding the Header in the Tags List, whenever there's an active subscription

File: Simplenote/StoreManager.swift [hunk 3]
```diff
@@ -91,11 +95,26 @@ class StoreManager {
         Task {
             do {
                 try await purchase(product: product)
+
+                Task { @MainActor in
+                    /// Note: This must be invoked in the main thread, otherwise, warnings get triggered!
+                    SPTracker.trackSustainerPurchaseCompleted(storeProduct: storeProduct)
```
Comment (bjhomer) [inline Simplenote/StoreManager.swift]: Then can we annotate `SPTracker` with `@MainActor`? That way, the compiler will enforce it for us!

**Review by bjhomer — APPROVED** [review]
This looks good. Thanks!

File: Simplenote/StoreManager.swift (hunk 3 above)
Comment (jleandroperez) [reply Simplenote/StoreManager.swift]: Done, thank you BJ!!

**Comment by jleandroperez** [comment]
Thanks again BJ!!

**Pull request merged**